### ISSUE TITLE
Bookmarks/ability to ‘save’ posts 

### ISSUE THREAD

**Issue by ChronicleFlask** [bluesky-social/social-app#1091]
**Is your feature request related to a problem? Please describe.**

I’d like to be able to ‘save’ posts to come back to later, without needing to publicly ‘like’ or reskeet them. 

**Describe the solution you'd like**

In Twitter it was the bookmark feature. I’d like something like that here. It was incredibly useful. 

**Describe alternatives you've considered**

Taking a screenshot and searching for the text string later. Which… is the clunkiest of solutions!

**Comment by mschwendener** [comment]
I agree

(as does https://bsky.app/profile/yilinwriter.bsky.social/post/3k4i46zmh5l2q )

**Comment by middleone20** [comment]
I also think this would be a great add. Besides Twitter/X, ActivityPub-based apps such as Mastodon also support this feature.

**Comment by danninov** [comment]
In surge of new Indonesian user this week, many user asking of bookmark feature. While there's workaround by using feed (like feed by Jaz), it would be nice to have native bookmark feature on bsky.

**Comment by saltedlolly** [comment]
Is this planned?

**Comment by tvanantwerp** [comment]
I'm curious what would be involved in creating this feature. Wouldn't it be possible to just duplicate the "Like" lexicons and app functionality? A bookmark is effectively the same thing, right? Just a user marking some post, and then being able to see a feed of those marked posts later.

**Comment by LovingNumbers** [comment]
> I'm curious what would be involved in creating this feature. Wouldn't it be possible to just duplicate the "Like" lexicons and app functionality? A bookmark is effectively the same thing, right? Just a user marking some post, and then being able to see a feed of those marked posts later.

then the bookmark would be public which most wouldn't like (likes on bluesky r public even when the app itself doesn't show it)

**Comment by edesdan** [comment]
I would love private bookmarks as well. Also bookmarking starter kit would be very cool feature.

**Comment by Apex-DE** [comment]
private Bookmarks are really missing within the app. Please implement.

**Comment by duanemoody** [comment]
Browsing the repo this looks like a matter of duplicating Likes except making it a method on repo to ensure it's private (and to speed up response time by not having to filter everyone's bookmarks for one user). I take it rkey is the post's unique ID within the poster's posts but not necessary a GUID (discussions suggest it enforces no particular scheme beyond length)?

**Comment by theAdhocracy** [comment]
Adding support, particularly if this outputs to an accessible feed. I use Mastodon's bookmark feature to effectively send content directly to Read Later apps like Readwise, Pocket, etc. Now looking for a way to replicate that behaviour with BlueSky, but currently not finding anything that works. This would hopefully solve that use case, too.

**Comment by AlexProber** [comment]
There is an external solution for this issue called Dewey: https://getdewey.co/
It is integrated within Bluesky and solves the problem for those that are interested in having private or public bookmarks.

**Comment by davidcelis** [comment]
I agree that Bluesky needs native support for bookmarks; this is a feature that people have come to expect from social networks like this. The 📌 feed is an interesting workaround, but [this post](https://bsky.app/profile/pfrazee.com/post/3lb4bx7iguk2b) from @pfrazee makes me worry about the _workarounds_ being what enter into the codebase for the social app rather than protocol-level support for bookmarks.

Looking through the lexicons, I'm not sure it's as simple as duplicating Likes except making it a method on `.repo`; as far as I can tell, most `com.atproto.repo` procedures do not require auth, including `com.atproto.repo.getRecord`. I may be misunderstanding whether or not repo records can truly be public or private, but it seems that all records are meant to be public.

It feels more like duplicating _mutes_ would work well for bookmarks, but I wasn't sure if there was any existing package within the `app.bsky` namespace that makes sense (mutes are in `app.bsky.graph`, but bookmarks aren't really a social graph concern). Maybe we need a new small `app.bsky.bookmark` package for this that follows the similar "require auth, only create and enumerate records for the actor" pattern present for mutes?

**Comment by duanemoody** [comment]
📌 is public (not good) and depends on a third party solution (worse). I apologize if I mischaracterized repo as being the user object

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ktotten** [comment]
Adding my vote to have this feature!

**Comment by davidcelis** [comment]
Given the possibility that this should be implemented in the protocol itself, I'd suggest folks weigh in on the discussion in the atproto repo instead of this social-app issue: https://github.com/bluesky-social/atproto/discussions/2000

**Comment by duanemoody** [comment]
Agreed on that, it should have been from the start. Thanks.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ebwinters** [comment]
The initial work at the lexicon and now client level is being done by myself and members of the lexicon-community org.
[here](https://github.com/lexicon-community/lexicon/pull/12) is the recently merged lexicon
[here](https://github.com/lexicon-community/js-lexicon-community-types/pull/2) is a draft of a client package for consumption in Bluesky or other app views

I could use some help doing the UI implementation in Bluesky, but the POC is done
![image](https://github.com/user-attachments/assets/eed61694-1ed7-48a8-8700-581c7143029f)

**Comment by danielbayley** [comment]
> If anyone on this thread can help with the UI work hit me up @western-red-cedar.bsky.social!

@ebwinters See https://github.com/bluesky-social/social-app/pull/7231#issuecomment-2557937960. Bookmark icons should be ready to go… I don't have time to delve into the codebase proper though, at the moment…

**Comment by ebwinters** [comment]
thanks @danielbayley! iconography is like half the battle in my mind. I don't know what design system they're using but it seems these fit

**Comment by mimonelu** [comment]
FYI: I have developed a similar bookmarking feature in [my personal client](https://klearsky.pages.dev) . You can try out the UX.
PDSls: https://pdsls.dev/at/did:plc:ilxxgyz7oz7mysber4omeqrg/space.aoisora.bookmark/3ldncin7co22o

![localhost_5173_(1440 x 900)](https://github.com/user-attachments/assets/c37e2f0b-4b50-4be6-8503-c4f74a46187c)

**Comment by SharpMars** [comment]
this looks like the main issue for this, so i wanted to say that when bookmarks get added it would be nice if bookmarks view could be paginated
this would make it easier to search through all of them, so for example it wouldnt require you to scroll all the way to get to some post that was bookmarked some time ago
so basically this would make things more convenient

**Comment by enderprism** [comment]
it would be nice for bookmarks to act like tags, to be able to have multiple categories of bookmarked posts

**Comment by cooljeanius** [comment]
> > > I'm curious what would be involved in creating this feature. Wouldn't it be possible to just duplicate the "Like" lexicons and app functionality? A bookmark is effectively the same thing, right? Just a user marking some post, and then being able to see a feed of those marked posts later.
> > 
> > 
> > then the bookmark would be public which most wouldn't like (likes on bluesky r public even when the app itself doesn't show it)
> 
> Yes: I want private bookmarks. We already have a form of public bookmarking in the form of feeds that look for a certain emoji response (I use one that looks for the pin). But that means everyone can see what I’ve bookmarked, and sometimes you really don’t want to broadcast that to everyone, for various reasons.

Also it can generate unwanted notifications for the people being replied to, as seen in [#8588](https://github.com/bluesky-social/social-app/issues/8588)

**Comment by mozzius** [comment]
Done in #8976!
